common/cache/lru.go

```diff
@@ -162,19 +233,19 @@ func (c *lru) Size() int {
 
 // Put puts a new value associated with a given key, returning the existing value (if present)
 // allowUpdate flag is used to control overwrite behavior if the value exists
-func (c *lru) putInternal(key string, value interface{}, allowUpdate bool) (interface{}, error) {
+func (c *lru) putInternal(key interface{}, value interface{}, allowUpdate bool) (interface{}, error) {
 	c.mut.Lock()
 	defer c.mut.Unlock()
 
 	elt := c.byKey[key]
 	if elt != nil {
-		entry := elt.Value.(*cacheEntry)
+		entry := elt.Value.(*entryImpl)
 		existing := entry.value
 		if allowUpdate {
 			entry.value = value
 		}
 		if c.ttl != 0 {
-			entry.expiration = time.Now().Add(c.ttl)
+			entry.timestamp = time.Now()
```

Comment:
Why are we change the expiration to timestamp?  i think expiration is better because it is more clear what it means.

Reply:
in terms of functionality, timestamp or expiration both works.
in terms of getting the last time a entry is refreshed, timestamp works, expiration do not. that is the reason